src/utils/utils.ts

```diff
@@ -13,25 +13,27 @@ import input from "../input.json";
 import { PrimitiveType } from "../types/AttributeType";
 
 export const getOrCreateDomainModel = async (
-  model: IModel
+  model: IModel,
+  moduleName: string
 ): Promise<domainmodels.DomainModel> => {
-  return await getDomainModelInterface(model).load();
+  return await getDomainModelInterface(model, moduleName).load();
 };
 
-const getDomainModelInterface = (model: IModel): domainmodels.IDomainModel => {
+const getDomainModelInterface = (model: IModel, moduleName: string): domainmodels.IDomainModel => {
   const existingDomainModelInterface = model
     .allDomainModels()
-    .filter((dm) => dm.containerAsModule.name === input.Module)[0];
+    .filter((dm) => dm.containerAsModule.name === moduleName)[0];
   if (existingDomainModelInterface) return existingDomainModelInterface;
-  else return createDomainModelInterface(model);
+  else return createDomainModelInterface(model, moduleName);
 };
 
 const createDomainModelInterface = (
-  model: IModel
+  model: IModel,
+  moduleName: string
 ): domainmodels.IDomainModel => {
   const project = model.allProjects()[0];
```

Comment:
Is this save? Do we indeed always need the first element in this array? If not, can we pass an identifier? If so, we could do something like this:

`const project = modle.allProducts().find(project => project.identifier === identifier);`